### PR TITLE
Mlir improvements

### PR DESCRIPTION
**Pull request by badumbatish**

### CODE REVIEW

File: shuriken/mjolnir/lib/transform/lifter.cpp [hunk 1]
```diff
@@ -42,21 +44,37 @@ using shuriken::parser::dex::DVMType;
...
+
+    // INFO: xrefs option disabled
+    analysis = std::make_unique<shuriken::analysis::dex::Analysis>(parser.get(), disassembler.get(), false);
+    auto mm = analysis->get_methods();
+    init();
+    mlir_gen_result = mlirGen();
+}
 void Lifter::init() {
+    registry.insert<::mlir::shuriken::MjolnIR::MjolnIRDialect>();
+    context.loadAllAvailableDialects();
```
Comment (jeppojeps) [inline shuriken/mjolnir/lib/transform/lifter.cpp:63]: do we really need all available dialects? Wouldn't it be an overkill?

**Review by jeppojeps — COMMENTED** [review]
it looks good to me

**Review by Fare9 — APPROVED** [review]

**Review by Fare9 — COMMENTED** [review]
Check in runtime if the instructions are generated correctly, you can use `_int.dex` to test if the arithmetic instructions are generated.

**Review by Fare9 — CHANGES_REQUESTED** [review]
Try `shuriken-opt` and check that the output generates the correct instructions.

**Review by Fare9 — APPROVED** [review]

**Pull request merged**